StrictEmit/ILGeneratorExtensions.Locals.cs

```diff
@@ -0,0 +1,38 @@
+﻿//
+//  ILGeneratorExtensions.Locals.cs
+//
+//  Copyright (c) 2018 Jarl Gullberg
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+using System;
+using System.Reflection;
+using System.Reflection.Emit;
+using JetBrains.Annotations;
+
+namespace StrictEmit
+{
+    public static partial class ILGeneratorExtensions
+    {
+        /// <summary>
+        /// Declares a local of type T.
+        /// </summary>
+        /// <typeparam name="T">The type.</typeparam>
+        /// <param name="il">The generator where the local is to be declared.</param>
+        /// <returns>The LocalBuilder declared with type T</returns>
+        [PublicAPI]
+        public static LocalBuilder DeclareLocal<T>([NotNull] this ILGenerator il) => il.DeclareLocal(typeof(T));
```

Comment:
This might be better placed in ILGeneratorExtensions.Variables.cs, since there's already local-related functions there.